tests/torchtune/datasets/test_chat_dataset.py

```diff
@@ -59,30 +23,44 @@ def dialogue(self):
         return [
             {
                 "dialogue": [
-                    {
-                        "role": "system",
-                        "content": "You are an AI assistant.",
-                        "masked": True,
-                    },
-                    {
-                        "role": "user",
-                        "content": "What is the meaning of life?",
-                        "masked": True,
-                    },
-                    {
-                        "role": "assistant",
-                        "content": "The meaning of life is 42.",
-                        "masked": False,
-                    },
-                    {"role": "user", "content": "That's ridiculous.", "masked": True},
-                    {"role": "assistant", "content": "I agree.", "masked": False},
+                    Message.from_dict(
+                        {
+                            "role": "system",
+                            "content": "You are an AI assistant.",
+                            "masked": True,
+                        }
+                    ),
+                    Message.from_dict(
+                        {
+                            "role": "user",
+                            "content": "What is the meaning of life?",
+                            "masked": True,
+                        }
+                    ),
+                    Message.from_dict(
+                        {
+                            "role": "assistant",
+                            "content": "The meaning of life is 42.",
+                            "masked": False,
+                        }
+                    ),
+                    Message.from_dict(
+                        {
+                            "role": "user",
+                            "content": "That's ridiculous.",
+                            "masked": True,
+                        }
+                    ),
+                    Message.from_dict(
+                        {"role": "assistant", "content": "I agree.", "masked": False}
+                    ),
                 ],
             },
         ]
 
     @mock.patch("torchtune.datasets._chat.load_dataset")
     def test_get_item(self, mock_load_dataset, chat_format, dialogue):
-        mock_load_dataset.return_value = Dataset.from_list(dialogue)
+        mock_load_dataset.return_value = dialogue
```

Comment:
Why this change? Shouldn't load_dataset be returning a `Dataset` object?

Reply:
I made the sample use Message objects, which Dataset.from_list doesn't like. Otherwise I need to write a quick `convert_to_messages`. I'm ok either way, this just seemed easier